From ae9bb04a693264138e0f4f583fcb13162eb3845c Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Tue, 28 Feb 2023 18:25:33 +0100
Subject: [PATCH] arm64: dts: st: add rpmsg irq controller for stm32mp257f-ev1

Add rpmsg IRQ controller support for communication with the Cortex-M33
Linux drivers can register generic interrupt to receive signals from the
remote processor.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: Ic344d45743644768728607efe46d6d8c3812b924
---
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
index 5d5d7c652455..d4eade16eb18 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -185,6 +185,16 @@ &m33_rproc {
 	status = "okay";
 };
 
+&mlahb {
+	intc_rpmsg: interrupt-controller@1 {
+		compatible = "rpmsg,intc";
+		reg = <1 0>;
+		#interrupt-cells = <1>;
+		interrupt-controller;
+		status = "okay";
+	};
+};
+
 &ommanager {
 	memory-region = <&mm_ospi1>;
 	memory-region-names = "mm_ospi1";
-- 
2.39.5

